### ISSUE TITLE
BLOCKER: How to seed randoms with entropy ?

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#2110]
This is a real BLOCKER but I hope easy to fix ? I need randoms that change from run-to-run. Our LPython random produces the same result every run.

```python
from random import random

print(random())  ########### PRODUCES SAME RESULT EVERY RUN ##########

# BUT, how to seed with time-of-day or internet noise
# or some other sort of variable entropy?

# NO WORK from random import seed
```

**Comment by certik** [comment]
As a quick fix, let's do:
```
from lpython import init_random
from random import random
init_random()
print(random())
```
Where the `init_random` function is our own function, doing nothing in CPython mode, but initializing the random number generator for LPython mode. That should unblock Brian.

Later we can figure out whether to call `init_random()` automatically in the main program and whether to always call it, or only if the user calls random, etc. That's lower priority.

**Comment by rebcabin** [comment]
Let init_random have an optional argument (or something like that) so I can seed the RNG for the cases where I do not want entropy, i.e., where I do want deterministic pseudorandoms.

**Comment by rebcabin** [comment]
Something that's more fitting is "seed"
https://docs.python.org/3/library/random.html#bookkeeping-functions

Even a partial implementation of that will be a great unblocker :)

**Comment by OmarMesqq** [comment]
Hello, I was thinking of grabbing this issue, and I already thought of a way of doing so with C++ `random`. I already have the code inside `libasr` in `rng.cpp`, but can't link it correctly. 

I already set the `SRC` in `CMakeLists.txt` to include my file and exported it wirh `extern "C" `, but still doesn't work.

Anyone got any tips?

**Comment by certik** [comment]
@OmarMesqq go ahead and submit a PR with what you have, and then we'll help improve it.

**Comment by certik** [comment]
I think @rebcabin is not blocked by this anymore, so I removed the BLOCKER tag.

**Comment by certik** [comment]
This issue will be fixed by https://github.com/lcompilers/lpython/pull/2318. 

There is another issue that the random numbers do not appear random, which is a separate issue: https://github.com/lcompilers/lpython/issues/2382.
